core/validation/validate.py

```diff
@@ -271,6 +272,7 @@ def validate_enums(
     :return: A list of InvalidEnumValueFailure objects for any rows with values outside
              the set of valid enum values.
     """
+    workbook[sheet_name]["original_index"] = workbook[sheet_name].index
     sheet = workbook[sheet_name].reset_index(drop=True)
```

Comment:
You can theoretically combine these on one line 
```sheet = workbook[sheet_name].reset_index()```
will insert the onld index as a column called Index